pkg/infrastructure/azure/azure.go

```diff
@@ -341,6 +341,7 @@ func (p *Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput
 		CustomerManagedKey: platform.CustomerManagedKey,
 		TokenCredential:    tokenCredential,
 		CloudConfiguration: cloudConfiguration,
+		AllowPublicAccess:  installConfig.PublicAPI(),
```

Comment:
Based on the terraform code 

https://github.com/openshift/installer/blob/d570c0561c0763f95b3eee6f80c7a84653db818a/data/data/azure/vnet/main.tf#L75

and [Jinyun's comment](https://issues.redhat.com/browse/OCPBUGS-37506?focusedId=25442517&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-25442517) it seems like this should be false unless there are customer managed keys, so perhaps:

```suggestion
		AllowPublicAccess:  platform.CustomerManagedKey != nil,
```